### PR TITLE
Bump argo-cd module from 1.8.7 -> 2.1.2

### LINKED ISSUE #462: Upgrade argo-cd dependency in go.mod from v1.8.7 to v2.1.2
### Is your task related to a problem? Please describe.
Currently, argocd-operator pulls in `v1.8.7` of argo-cd as a dependency in go.mod, which is an old version of argo-cd

### Describe the solution you'd like
We should update the go.mod file to pull in argo-cd `v2.1.2` as a dependency instead

### PR DESCRIPTION
**Pull request by tylerauerbeck**
**What type of PR is this?**
/kind enhancement

**What does this PR do / why we need it**:

This bumps the argo-cd dependency from 1.8.7 to 2.1.2

**Have you updated the necessary documentation?**

* [X] Documentation update is required by this PR.
* [X] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #462 ?

**How to test changes / Special notes to the reviewer**:

Run existing tests to ensure that all existing tests continue to pass (according to docs):
- `make test`
- ARGOCD_CLUSTER_CONFIG_NAMESPACES=argocd-e2e-cluster-config make install run
- hack/test.sh

### CODE REVIEW

File: openshift.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build openshift
```
Comment (iam-veeramalla) [inline openshift.go:1]: @tylerauerbeck is this change intentional ?

Comment (tylerauerbeck) [reply openshift.go]: @iam-veeramalla Fixed.

**Comment by iam-veeramalla** [comment]
@tylerauerbeck Thanks for the PR :)

**Review by iam-veeramalla — APPROVED** [review]
